pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -326,6 +331,14 @@ func (s *NodeController) checkNodeReady(node *api.Node) *api.NodeCondition {
 	}
 }
 
+// sendConditionEvent sends condition change event.
+func (s *NodeController) sendConditionEvent(node *api.Node, kind api.NodeConditionKind, oldCondition, newCondition *api.NodeCondition) {
+	// TODO: if 'oldCondition' is nil, we should publish the initial condition, but selfLink is not set.
+	if oldCondition != nil {
+		record.Eventf(node, "condition change", "Node condition %v changes from %v to %v", kind, oldCondition.Status, newCondition.Status)
```

Comment:
Oh, lavalamp, long time no see :)

By docs, I guess you mean the comment above func Event().  I didn't notice it before.  But I definitely looked at how other component send event, there is "failedScheduling" in scheduler, "failedValidation" in kubelet.  So I guess "changedCondition"?

Actually, the fact that reason field is a string doesn't looks like it will be switched on.  Maybe we can change them to first class type in k8s.